crates/sui-protocol-config/src/lib.rs

```diff
@@ -2069,6 +2070,48 @@ impl ProtocolConfig {
         cfg
     }
 
+    // Extract the bytecode verifier config from this protocol config. `for_signing` indicates
+    // whether this config is used for verification during signing or execution.
+    pub fn verifier_config(&self, for_signing: bool) -> VerifierConfig {
+        let (max_back_edges_per_function, max_back_edges_per_module) = if for_signing {
+            (
+                Some(self.max_back_edges_per_function() as usize),
+                Some(self.max_back_edges_per_module() as usize),
+            )
+        } else {
+            (None, None)
```

Comment:
Do we need this sort of split? I thought we used the DummyMeter outside of signing (in which case these would be ignored?)

Reply:
The problem is that these configs are not part of metering, they are part of the verifier itself. I do think it's a bit incongruous that we change these verifier configs to be more restrictive during signing, but I didn't want to remove them.

Previously we were using the same `Meter` whether it was signing or not, and had to pass it different parameters. This PR changes that requirement, so it highlights this incongruity a bit more. I think it's something we can definitely revisit -- it would clean things up even more.